### PR TITLE
Port mainapp caelum rc4

### PR DESCRIPTION
**Pull request by NathanaelGandhi**

### CODE REVIEW

**Comment by LucienMorey** [comment]
I modified the path of trans_udp.h locally in to_custom.c to get this to compile. I expect this to compile properly once that has been ported successfully as well.

File: fsw/src/to_custom.c [hunk 1]
```diff
@@ -99,34 +100,39 @@ int32 TO_CustomInit(void)
...
+       here and update the TO_NUM_CRITICAL_MIDS value. */
 
     /* Set Critical Message Ids which must always be in config table. */
     g_TO_AppData.criticalMid[0] = TO_HK_TLM_MID;
     g_TO_AppData.criticalMid[1] = CI_HK_TLM_MID;
-    g_TO_AppData.criticalMid[2] = CFE_EVS_EVENT_MSG_MID; 
-    
+    // TODO TO_LAB has this as LONG_EVENT so I am taking that in the interest of expedience. This could also be
+    // CFE_EVS_SHORT_EVENT_MSG_MID or something totally different... -LM
+    g_TO_AppData.criticalMid[2] = CFE_EVS_LONG_EVENT_MSG_MID;
```
Comment (LucienMorey) [inline fsw/src/to_custom.c:110]: Needs to be documented somewhere

File: fsw/src/to_custom.c [hunk 2]
```diff
@@ -99,34 +100,39 @@ int32 TO_CustomInit(void)
...
-int32 TO_CustomAppCmds(CFE_SB_Msg_t* pMsg)
+int32 TO_CustomAppCmds(const CFE_SB_Buffer_t *pMsg)
 {
-    int32 iStatus = TO_SUCCESS;
-    uint32 uiCmdCode = CFE_SB_GetCmdCode(pMsg);
+    int32             iStatus   = TO_SUCCESS;
+    CFE_MSG_FcnCode_t uiCmdCode = 0;
+
+    // TODO Potentially modify this to return real status code from getFcnCode instead of just breaking on -1 -LM
+    CFE_MSG_GetFcnCode(&pMsg->Msg, &uiCmdCode);
```
Comment (LucienMorey) [inline fsw/src/to_custom.c:129]: Needs to be documented somewhere

File: fsw/tables/to_config.c [hunk 3]
```diff
@@ -56,155 +56,159 @@ extern "C" {
...
-    }
-};
+TO_ConfigTable_t to_ConfigTable = {
+    {/* 0 - 9 */
+     {CF_CONFIG_TLM_MID, {0, 0}, 1, 0xffff, TO_GROUP_CFE | TO_MGROUP_ONE, 0, 1},
+     {CF_HK_TLM_MID, {0, 0}, 1, 0xffff, TO_GROUP_CFE | TO_MGROUP_ONE, 0, 1},
+     //  TODO I dont think this functionality exists with the update to CF 3.0.0. If I am interpreting the statement
+     //  "Much smaller code footprint. CF 3.0 is light-weight flight-only app. It does not provide any ground engine
+     //  support." correctly, then this cna be commented out and not used. - LM
+     //  {CF_SPACE_TO_GND_PDU_MID, {0, 0}, 32, 0xffff, TO_GROUP_CFE | TO_MGROUP_ONE, 0, 1},
```
Comment (LucienMorey) [inline fsw/tables/to_config.c:81]: Needs to be documented somewhere

File: fsw/tables/to_config.c [hunk 4]
```diff
@@ -56,155 +56,159 @@ extern "C" {
...
+     {TO_UNUSED_ENTRY, {0, 0}, 0, 0x0000, TO_GROUP_NONE, 0, 0},
+
+     // TODO I dont think there are notes about this MID removal. It could be related to not supporting ground stations
+     // anymore. making unused for now - LM
+     //  {CF_TRANS_TLM_MID, {0, 0}, 1, 0xffff, TO_GROUP_CFE | TO_MGROUP_ONE, 0, 1},
```
Comment (LucienMorey) [inline fsw/tables/to_config.c:86]: Needs to be documented somewhere

File: fsw/tables/to_config.c [hunk 5]
```diff
@@ -56,155 +56,159 @@ extern "C" {
...
+     {TO_UNUSED_ENTRY, {0, 0}, 0, 0x0000, TO_GROUP_NONE, 0, 0},
+
+     {CFE_ES_APP_TLM_MID, {0, 0}, 1, 0xffff, TO_GROUP_CFE | TO_MGROUP_ONE, 0, 1},
+     {CFE_ES_HK_TLM_MID, {0, 0}, 1, 0xffff, TO_GROUP_CFE | TO_MGROUP_ONE, 0, 1},
+     {CFE_ES_MEMSTATS_TLM_MID, {0, 0}, 1, 0xffff, TO_GROUP_CFE | TO_MGROUP_ONE, 0, 1},
+     //  This has been deprecated according to issue nasa/to_lab#54. replacing with unsused for now until we can verify
+     //  a replacement if there is one -LM
+     // {CFE_ES_SHELL_TLM_MID, {0, 0}, 32, 0xffff, TO_GROUP_CFE | TO_MGROUP_ONE, 0, 1},
```
Comment (LucienMorey) [inline fsw/tables/to_config.c:94]: Needs to be documented somewhere

File: fsw/tables/to_config.c [hunk 6]
```diff
@@ -56,155 +56,159 @@ extern "C" {
...
+     {TO_UNUSED_ENTRY, {0, 0}, 0, 0x0000, TO_GROUP_NONE, 0, 0},
+     // TODO TO_LAB has this as LONG_EVENT so I am taking that in the interest of expedience. This could also be
+     // CFE_EVS_SHORT_EVENT_MSG_MID or something totally different... -LM
+     // {CFE_EVS_EVENT_MSG_MID, {0, 0}, 32, 0xffff, TO_GROUP_CFE | TO_MGROUP_ONE, 0, 1},
```
Comment (LucienMorey) [inline fsw/tables/to_config.c:98]: Needs to be documented somewhere

**Review by LucienMorey — APPROVED** [review]

File: fsw/src/to_custom.c (hunk 2 above)
Comment (NathanaelGandhi) [reply fsw/src/to_custom.c:129]: https://github.com/Space-Machines-Company/TO-SMC/issues/2

File: fsw/tables/to_config.c (hunk 5 above)
Comment (NathanaelGandhi) [reply fsw/tables/to_config.c:94]: https://github.com/Space-Machines-Company/TO-SMC/issues/4

File: fsw/tables/to_config.c (hunk 6 above)
Comment (NathanaelGandhi) [reply fsw/tables/to_config.c:98]: https://github.com/Space-Machines-Company/TO-SMC/issues/5

File: fsw/src/to_custom.c (hunk 1 above)
Comment (NathanaelGandhi) [reply fsw/src/to_custom.c:110]: https://github.com/Space-Machines-Company/TO-SMC/issues/2

**Pull request merged**